File: src/Entity/Destination/Favorite.php
<?php

namespace App\Entity\Destination;


use Doctrine\ORM\Mapping as ORM;
use App\Entity\User;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

#[ORM\Entity]
class Favorite
{

    #[ORM\ManyToOne(targetEntity: Destination::class)]
    #[ORM\JoinColumn(name: 'id_destination', referencedColumnName: 'id_destination')]
    private ?Destination $id_destination = null;
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private int $id_favorite;

     #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'id_user', referencedColumnName: 'id', nullable: false)]
    private ?User $id_user = null;


    public function getIdfavorite(): int
    {
        return $this->id_favorite;
    }

    public function setIdfavorite(int $id_favorite): self
    {
        $this->id_favorite = $id_favorite;
        return $this;
    }


public function setIdDestination(?Destination $idDestination): self
{
    $this->id_destination = $idDestination;
    return $this;
}


    public function getIduser(): ?User
    {
        return $this->id_user;
    }

    public function setIdUser(?User $idUser): self
{
    $this->id_user = $idUser;
    return $this;
}



    public function getDestination(): ?Destination
{
    return $this->id_destination;
}

}

File: src/main/java/Models/Favorite.java
package Models;

public class Favorite {
    private int id_favorite;
    private int id_user;
    private int id_destination;


    public Favorite( int id_favorite,int id_user, int id_destination) {
        this.id_favorite = id_favorite;
        this.id_user = id_user;
        this.id_destination = id_destination;
    }

    public Favorite( int id_user, int id_destination) {
        this.id_user = id_user;
        this.id_destination = id_destination;
    }


    public int getId_favorite() {
        return id_favorite;
    }
    public void setId_favorite(int id_favorite) {
        this.id_favorite = id_favorite;
    }
    public int getId_user() {
        return id_user;
    }
    public void setId_user(int id_user) {
        this.id_user = id_user;
    }
    public int getId_destination() {
        return id_destination;
    }
    public void setId_destination(int id_destination) {
        this.id_destination = id_destination;
    }

}
